Editor Changes for Role landing pages for issue #87

OHaiDanny opened the pull request:
@nonowazu in this PR i've added the editor changes for roles. I hope I understood the task correctly.. #87 

Couple clarifying things:

1. The names of the roles should be in plural form?
2. the role for melee, ranged, caster dps should be identifed as `dps`?

nonowazu commented on exampleSite/content/jobs/ranged/_index.md:
```diff
@@ -0,0 +1,8 @@
+  main:
+    name: Ranged
+    parent: jobs
+role: dps
```
I actually think this is `role: ranged` – at least, that's what I've got working with production content. Double-check with @Roxas92 and @ScrubbyXIV though

OHaiDanny replied:
Ok let me take it away and check

nonowazu commented on exampleSite/static/admin/config.yml:
```diff
@@ -50,6 +50,70 @@ collections:
+          default:
+            main:
+              name: "Melees"
+              parent: "jobs"
+        - label: "Role"
+          name: "role"
+          widget: "hidden"
+          default: "melee"
+      - label: "Ranged"
+        name: "ranged"
+        file: "content/jobs/ranged/_index.md"
+        fields:
+        - {label: "Body", name: "body", widget: "markdown"}
+        - label: "Menu hierarchy"
+          name: "menu"
+          widget: "hidden"
+          default:
+            main:
+              name: "Ranged"
+              parent: "jobs"
+        - label: "Role"
+          name: "role"
+          widget: "hidden"
+          default: "ranged"
+      - label: "Casters"
+        name: "casters"
+        file: "content/jobs/casters/_index.md"
+        fields:
+        - {label: "Body", name: "body", widget: "markdown"}
+        - label: "Menu hierarchy"
+          name: "menu"
+          widget: "hidden"
+          default:
+            main:
+              name: "Casters"
+              parent: "jobs"
+        - label: "Role"
+          name: "role"
+          widget: "hidden"
+          default: "caster"
```
Just to make 100% sure – you used the editor to generate the other `_index.md` files as part of this PR?

OHaiDanny replied:
Yes that is correct :)

OHaiDanny replied:
actually i'm confusing myself a bit with the Job landing page one. Scratch my previous comment. I didn't check it. Let me do that now.

OHaiDanny replied:
ugh these hidden labels again. Let me try to set them up so that they are visible

nonowazu replied:
They're tricky beasts. That's why I thought just going with selections with the value hardcoded might be best.

OHaiDanny replied:
Yep! i've turned them into selections in my recent updates. Feel free to review it again @nonowazu

OHaiDanny commented:
@nonowazu I've updated it so that a brand new role being added and saved in the CMS will generate the heirarchy in each `_index.md`.

Also adjusted the role name after chatting with Scrubby.

nonowazu commented on exampleSite/content/jobs/melee/_index.md:
```diff
@@ -0,0 +1,8 @@
+    parent: melee
+role: melee
+---
+Melee do positional and stuff.
```
uhhh, I notice there's a `melee/_index.md` and `melees/_index.md` here.

OHaiDanny replied:
oh shoot! fixing it right away

OHaiDanny replied:
prolly was generated when I was testing and I forgot to delete it.

OHaiDanny replied:
Ok I removed `melees` since we are going with `melee`

nonowazu reviewed (changes requested):
I think we're good, just a minor, *minor* nit
